File: app/src/main/java/com/example/newsapp/repository/NewsRepository.java
package com.example.newsapp.repository;

import com.example.newsapp.api.ArticleNewsService;
import com.example.newsapp.model.ResultNews;

import io.reactivex.rxjava3.core.Single;

public class NewsRepository {
    private ArticleNewsService articleNewsService;

    public NewsRepository(ArticleNewsService articleNewsService) {
        this.articleNewsService = articleNewsService;
    }
    public Single<ResultNews> getArticleNews(String apiKey, String category, String country) {
        return articleNewsService.getAllArticleNews(apiKey, category, country);
    }
}

File: app/src/main/java/com/example/newsapp/repository/NewsRepository.kt
package com.example.newsapp.repository

import androidx.paging.PagingData
import com.example.newsapp.model.ArticleNews
import kotlinx.coroutines.flow.Flow

interface NewsRepository {
    fun getNews(category: String, country: String): Flow<PagingData<ArticleNews>>

    fun searchArticle(query: String, country: String): Flow<PagingData<ArticleNews>>

    val getSavedArticleNews: Flow<List<ArticleNews>>

    fun getSavedArticleNewsByTitle(title: String) : Int

    fun saveArticle(articleNews: ArticleNews)

    fun deleteArticle(title: String)
}